src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/dynamicProgramming/DynamicProgramming.java

```diff
@@ -0,0 +1,74 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.dynamicProgramming;
+
+import se.liu.ida.hefquin.engine.federation.FederationAccessException;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpJoin;
+import se.liu.ida.hefquin.engine.queryplan.physical.BinaryPhysicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.*;
+import se.liu.ida.hefquin.engine.queryproc.*;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.utils.CardinalityEstimation;
+
+import java.util.*;
+
+public class DynamicProgramming extends CardinalityEstimation {
```

Comment:
It doesn't make sense to make this class a subclass of CardinalityEstimation. Instead, pass a CardinalityEstimation object to the constructor of this class. This way, we can use different implementations of CardinalityEstimation in the future.